.github/workflows/python-packaging.yml

```diff
@@ -98,110 +84,81 @@ jobs:
           CIBW_ARCHS_MACOS: ${{ matrix.cibw_archs_macos }}
           CIBW_TEST_SKIP: "*_arm64"
           CMAKE_OSX_ARCHITECTURES: ${{ matrix.cibw_archs_macos }}
-      # See https://github.com/single-cell-data/TileDB-SOMA/pull/2620
-      #
-      # * MacOS 12 and 14 self-report as `macosx-10.9-universal2` via
-      #   `distutil.util.get_platform()`.
-      # * Due to https://github.com/single-cell-data/TileDB-SOMA/pull/2124 we build
-      #   with `-mmacosx-version-min=11.0` which overrides any attempt at
-      #   setting `CMAKE_OSX_DEPLOYMENT_TARGET` in our env here.
-      # * This means a MacOS 12/14 system can self-report as 10.9, see
-      #   a wheel at 11.0, and refuse the install, claiming that no suitable
-      #   OS-version support can be found in the wheel.
-      # Therefore we rename the wheel's reported supported MacOS version to
-      # match the self-reported OS version of today's clients.
-      #
-      # Example file names we assume here:
-      # tiledbsoma-VERSIONHERE-cp311-cp311-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-      # tiledbsoma-VERSIONHERE-cp311-cp311-macosx_14_0_x86_64.whl
-      # tiledbsoma-VERSIONHERE-cp311-cp311-macosx_14_0_arm64.whl
-      - name: MacOS portability step
-        run: |
-          pwd
-          ls -l
-          cd wheelhouse
-          ls -l
-          for wheel_file_name in *.whl; do
-            if expr "$wheel_file_name" : ".*macosx_11_0.*"; then
-              new_name=$(echo $wheel_file_name | sed "s/macosx_11_0/macosx_10_9/")
-              if [ "$wheel_file_name" = "$new_name" ]; then
-                echo "Failed to rename $wheel_file_name"
-                exit 1
-              fi
-              echo Renaming $wheel_file_name to $new_name
-              mv $wheel_file_name $new_name
-            fi
-          done
-          cd ..
-          echo
-      - name: Upload wheels-${{ matrix.wheel_name }} to GitHub Actions storage
-        uses: actions/upload-artifact@v3
+      - name: Upload wheel-${{ matrix.wheel-name }}-${{ matrix.python-version }} to GitHub Actions storage
+        uses: actions/upload-artifact@v4
         with:
-          name: wheels-${{ matrix.wheel_name }}
+          name: wheel-${{ matrix.wheel-name }}-${{ matrix.python-version }}
           path: ./wheelhouse/*.whl
 
   # This step locally tries out the built wheels, without publishing to PyPI
   smoke-test:
-    name: Smoke test wheels
+    name: "Smoke test ${{ matrix.python.v }}-${{ matrix.wheel-name }} wheel"
     needs: wheels
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        python:
+          - v: '310'
+            vv: '3.10'
```

Comment:
Please find some names more descriptive than `v` and `vv`

Maybe `undotted_version` and `dotted_version`